cli/command/cli.go

```diff
@@ -195,8 +197,24 @@ func isEnabled(value string) (bool, error) {
 }
 
 func (cli *DockerCli) initializeFromClient() {
-	ping, err := cli.client.Ping(context.Background())
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
Does that imply modifying every context creation from `background` to now be created from this new context? Should it be a global too like background ? Or be passed as an argument everywhere? My point is, it can be a huge refactor to introduce this top level context, so maybe we should introduce it in another PR then use it in this one. WDYT @cpuguy83 ?

Reply:
Making the change in this PR seems pretty minimal if we are only doing it for this function.
We can spread that context around to other functions as a separate PR but I think this is a great place to get started from.